src/app/components/osd/osd.component.ts

```diff
@@ -96,16 +117,25 @@ export class OsdComponent implements AfterViewInit, OnDestroy {
   get options() { return this._options; }
   optionsChange = new BehaviorSubject({});
 
-  // tslint:disable-next-line: variable-name
-  private _manifestURL: string;
-  @Input() set manifestURL(v: string) {
-    if (v !== this._manifestURL) {
-      this._manifestURL = v;
-      this.manifestURLChange.next(this._manifestURL);
+  private _viewerDataType: string; // tslint:disable-line: variable-name
+  private _manifestPath: string; // tslint:disable-line: variable-name
+  private _xmlImagesPath: XMLImagesValues[]; // tslint:disable-line: variable-name
+  @Input() set data(v: ViewerDataType) {
+    if (v.value.manifestURL !== this._manifestPath) {
+      this._manifestPath = v.value.manifestURL;
+      this.manifestChange.next(this._manifestPath);
     }
+    if (v.value.xmlImages !== this._xmlImagesPath) {
+      this._xmlImagesPath = v.value.xmlImages;
+      this.xmlImageChange.next(this._xmlImagesPath);
+    }
+    this._viewerDataType = v.type;
   }
-  get manifestURL() { return this._manifestURL; }
-  manifestURLChange = new BehaviorSubject(undefined);
+  get viewerDataType() { return this._viewerDataType; }
+  get manifestURL() { return this._manifestPath; }
+  get xmlImageURL() { return this._xmlImagesPath; }
```

Comment:
Same as `viewerDAtaType`, if you want to access these members why do you keep them private?